Makefile

```diff
@@ -284,6 +302,41 @@ TEST_CC_CODE_BINS := $(SINGLE_FILE_CC_PROGRAM_BIN)
 C_UNIT_TEST_BIN     := $(SINGLE_FILE_C_PROGRAM_BIN)
 CPP_UNIT_TEST_BIN   := $(SINGLE_FILE_CPP_PROGRAM_BIN)
 CC_UNIT_TEST_BIN    := $(SINGLE_FILE_CC_PROGRAM_BIN)
+L3_C_UNIT_TEST_BIN  := $(BINDIR)/$(UNIT_DIR)/l3_dump.py-test
+
+# ##############################################################################
+# Generate symbols and dependencies to build unit-test sources
+# ##############################################################################
+
+UNIT_TESTBIN_SRCS := $(filter %-test.c, $(UNIT_TESTSRC))
+UNIT_TESTBINS     := $(UNIT_TESTBIN_SRCS:$(TESTS_DIR)/%-test.c=$(BINDIR)/%-test)
+ifeq "$(BUILD_VERBOSE)" "1"
+    $(info )
+    $(info ---- Debug ----)
+    $(info $$UNIT_DIR = [ ${UNIT_DIR} ])
+    $(info $$UNITTESTS_DIR = [ ${UNITTESTS_DIR} ])
+    $(info $$UNIT_TESTBINS = [ ${UNIT_TESTBINS} ])
+    $(info )
+endif
+
+# All the stand-alone unit-tests depend on the core L3 file, so add those objects, too.
+# Given <x>, this macro generates a line of Makefile of the form
+# bin/unit/<x>: obj/unit/<x>.o
+define unit_test_self_dependency =
+$(1): $(patsubst $(BINDIR)/$(UNIT_DIR)/%,$(OBJDIR)/$(UNITTESTS_DIR)/%.o, $(1)) \
+        $(L3_OBJS)
+endef
+
+# ---------------------------------------------------------------------------------
+# This invocation will generate makefile-snippets specifying the dependency of
+# each unit-test binary on its sources and required objects. This way, we do not
+# have to enumerate dependency for each unit-test binary built, such as:
+#
+# $(BINDIR)/$(UNIT_DIR)/l3_dump.py-test: $(OBJDIR)/$(UNITTESTS_DIR)/l3_dump.py-test.o \
+#                                        $(OBJDIR)/$(SRCDIR)/l3.o
+# See https://www.gnu.org/software/make/manual/html_node/Eval-Function.html
+# ---------------------------------------------------------------------------------
+$(foreach unit,$(UNIT_TESTBINS),$(eval $(call unit_test_self_dependency,$(unit))))
```

Comment:
For now, the invocation of the make-helper function `unit_test_self_dependency()` will generate the required dependency as shown on L335.

In future, if we add more unit-test programs, their dependencies will be automatically generated by this clause. So, there should be nothing more to do in the future in this local area.